### PR TITLE
Kubernetes network policy

### PR DESCRIPTION
**Pull request by bmckercher123**
## Todos

- [ ] Tests
- [ ] Documentation
- [ ] Release note

## Release Note

```release-note
None required
```

### CODE REVIEW

**Comment by caseydavenport** [comment]
Deploy preview for *calico* ready!

Built with commit 2ad91fb177e9ee91c38f734ff48f701c230acde4

https://deploy-preview-2489--calico.netlify.com

File: v3.6/security/kubernetes-network-policy.md [hunk 1]
```diff
@@ -0,0 +1,178 @@
...
+The Kubernetes Network Policy API provides a standard way for users to define network policy for controlling network connections. However, Kubernetes has no built-in capability to enforce the network policy. To enforce network policy, you must use a network plugin such as Calico. 
+
+#### Ingress and egress
+
+The bulk of securing network connections typically revolves around defining egress and ingress rules. From the point of view of a Kubernetes pod, **ingress** is incoming connections to the pod, and **egress** is outgoing connections from the pod. In Kubernetes network policy, you create ingress and egress “allow” rules independently (egress, ingress, or both). 
+
+#### Default deny/allow behavior
+
+**Default allow** means all connections are allowed by default, unless otherwise specified. 
+**Default deny** means all connections are denied by default, unless explicitly allowed.
```
Comment (ozdanborne) [inline v3.6/security/kubernetes-network-policy.md]: FYI you've put these on 2 separate lines, but they will appear on the same line when rendered into html. If you'd like them to appear on separate lines, put an additional newline between them.

File: v3.6/security/kubernetes-network-policy.md [hunk 2]
```diff
@@ -0,0 +1,178 @@
...
+
+<pre>
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: allow-same-namespace
+  namespace: default
+spec:
+  podSelector:
+    matchLabels: {“<b>color</b>”: “<b>blue</b>”}
```
Comment (ozdanborne) [inline v3.6/security/kubernetes-network-policy.md]: While this is a valid networkpolicy yaml, we usually show dictionaries (`{}`)  in a different way, like this:

```
spec:
  podSelector:
    matchLabels:
      <b>color</b>: <b>blue</b>
```

The quotes `"` aren't necessary either.

They both are interpreted the same way, so not a huge deal either way. I'd argue that my suggested way looks a little nicer...

File: v3.6/security/kubernetes-network-policy.md [hunk 3]
```diff
@@ -0,0 +1,186 @@
...
+  name: allow-same-namespace
+  namespace: default
+spec:
+  podSelector:
+    matchLabels:
+      color: blue
+  ingress:
+  - from:
+    - podSelector:
+      <b>color</b>: <b>red</b>
```
Comment (ozdanborne) [inline v3.6/security/kubernetes-network-policy.md]: This needs to be indented 2 more spaces
```suggestion
        <b>color</b>: <b>red</b>
```

File: v3.6/security/kubernetes-network-policy.md [hunk 4]
```diff
@@ -0,0 +1,186 @@
...
+  name: allow-same-namespace
+  namespace: default
+spec:
+  podSelector:
+    matchLabels: 
+      <b>color</b>: <b>blue</b>
+  ingress:
+  - from:
+    - podSelector: 
+      <b>color</b>: <b>red</b>
```
Comment (ozdanborne) [inline v3.6/security/kubernetes-network-policy.md]:
```suggestion
        <b>color</b>: <b>red</b>
```

File: v3.6/security/kubernetes-network-policy.md [hunk 5]
```diff
@@ -0,0 +1,186 @@
...
+  name: allow-egress-same-namespace
+  namespace: default
+spec:
+  podSelector:
+    matchLabels:
+      color: blue
+  egress:
+  - to:
+    - podSelector: 
+      <b>color</b>: <b>red</b>
```
Comment (ozdanborne) [inline v3.6/security/kubernetes-network-policy.md]:
```suggestion
        <b>color</b>: <b>red</b>
```

**Comment by caseydavenport** [comment]
Merged in https://github.com/projectcalico/calico/pull/2493

:tada:

**Pull request closed without merging**